spec/app/ics-101-interchain-swap/README.md

```diff
@@ -724,514 +843,674 @@ interface MsgSwapResponse {
-        stateChange: {
-            poolToken: msg.poolToken,
-            out: [outToken],
-        }
-    }
-    sendInterchainIBCSwapDataPacket(packet, msg.sourcePort, msg.sourceChannel, msg.timeoutHeight, msg.timeoutTimestamp)
+    const { pool, found } = await store.getInterchainLiquidityPool(msg.PoolId);
+    abortTransactionUnless(found)
+
+    abortTransactionUnless(pool.SourceChainId !== store.ChainID())
+    abortTransactionUnless(pool.DestinationCreator === msg.Creator)
+
+    const creatorAddr = sdk.MustAccAddressFromBech32(msg.Creator);
+
+    const asset = pool.FindAssetBySide("SOURCE");
+    abortTransactionUnless(asset)
+
+    const liquidity = store.GetBalance(creatorAddr, asset.denom);
+    abortTransactionUnless(liquidity.amount > 0)
+
+
+    const lockErr = store.LockTokens(pool.counterPartyPort, pool.counterPartyChannel, creatorAddr, asset);
+    abortTransactionUnless(lockErr === undefined)
+
+    const packet: IBCSwapPacketData = {
+      type: "TAKE_POOL",
+      data: protobuf.encode(msg),
+    };
+
+
+    const sendPacketErr = await store.SendIBCSwapPacket(pool.counterPartyPort, pool.counterPartyChannel, timeoutHeight, timeoutStamp, packet);
+    abortTransactionUnless(sendPacketErr === undefined)
+
+    return {
+      poolId: msg.PoolId,
+    };
+  }
+
+  function makeMultiAssetDeposit(msg: MsgMakeMultiAssetDepositRequest): MsgMultiAssetDepositResponse {
+
+    const { pool, found } = await store.getInterchainLiquidityPool(msg.poolId);
+
+    abortTransactionUnless(found)
+
+    // Check initial deposit condition
+    abortTransactionUnless(pool.status === "ACTIVE")
+
+    // Check input ratio of tokens
+    const sourceAsset = pool.findAssetBySide("SOURCE");
+    abortTransactionUnless(sourceAsset)
+
+    const destinationAsset = pool.findAssetBySide("DESTINATION");
+    abortTransactionUnless(destinationAsset)
+
+    const currentRatio = sourceAsset.amount.Mul(sdk.NewInt(1e18)).Quo(destinationAsset.amount);
+    const inputRatio = msg.deposits[0].balance.amount.Mul(sdk.NewInt(1e18)).Quo(msg.deposits[1].balance.amount);
+
+    const slippageErr = checkSlippage(currentRatio, inputRatio, 10);
+    abortTransactionUnless(slippageErr)
+
+    // Create escrow module account here
+    const lockErr = store.lockTokens(pool.counterPartyPort, pool.counterPartyChannel, sdk.msg.deposits[0].sender, msg.deposits[0].balance);
+    abortTransactionUnless(lockErr)
+
+    const amm = new InterchainMarketMaker(pool);
+
+    const poolTokens = await amm.depositMultiAsset([
+      msg.deposits[0].balance,
+      msg.deposits[1].balance,
+    ]);
+
+    // create order
+    const order: MultiAssetDepositOrder = {
+      poolId: msg.poolId;
+      chainId: store.chainID(),
+      sourceMaker: msg.deposits[0].sender,
+      destinationTaker: msg.deposits[1].sender,
+      deposits: getCoinsFromDepositAssets(msg.deposits),
+      status: "PENDING";
+      createdAt: store.blockHeight(),
```

Comment:
using blockTime is more accurate.

Reply:
To use blockTime is accurate but there are possibility to manipulate by validator, so most of case, prefer to use block height instead of blockTimeStamp.